### DISCUSSION TITLE
Typescript: Partial Form accepting generic Form return

### DISCUSSION THREAD

**Discussion by Joe312341** [react-hook-form/react-hook-form#7199] [Q&A]
Hello,

I know this was discussed here: https://github.com/react-hook-form/react-hook-form/issues/6726

The problem is that because of the limitations of `Path` not correctly dealing with generics it is not possible to have a partial form with a set up such as 

```
interface PartialFormValues {
  name: string;
} 

interface PartialFormProps<FormValues> {
  form: UseFormReturn<FormValues>;
}

const PartialForm = <FormValues extends PartialFormValues>({
form
}: PartialFormProps<FormValues>): JSX.Element => { ... })
```

This works fine for `getValues`, but will throw an error when you use `watch`, `setValue` etc, because they all involve `Path` type.

The suggestion given in the thread was to have the main form only have optional attributes and then set up the PartialForm so that it accepts its own form values

```
interface PartialFormValues {
  name: string;
} 

interface MainFormValues extends PartialFormValues {
 age?: number; 
}

interface PartialFormProps {
  form: UseFormReturn<PartialFormValues>;
}

const PartialForm = ({
form
}: PartialFormProps): JSX.Element => { ... })
```

This works, but with the limitation of having to make all the attributes of the main form optional.

Is there any other way around this?
Thank you

**Comment by bluebill1049** [comment]
FYI: We are looking into this issue.
